File: builder/SlackNotificationBuilder.java
package builder;

public class SlackNotificationBuilder implements NotificationBuilder {
    private String content;
    private String sender;
    private String recipient;
    private String timestamp;
    private boolean hasMarkdownLanguage;

    @Override
    public SlackNotificationBuilder setContent(String content) {
        this.content = content;
        return this;
    }

    @Override
    public SlackNotificationBuilder setSender(String sender) {
        this.sender = sender;
        return this;
    }

    @Override
    public SlackNotificationBuilder setRecipient(String recipient) {
        this.recipient = recipient;
        return this;
    }

    @Override
    public SlackNotificationBuilder setTimestamp(String timestamp) {
        this.timestamp = timestamp;
        return this;
    }

    public SlackNotificationBuilder setMarkdownLanguage(boolean hasMarkdownLanguage) {
        this.hasMarkdownLanguage = hasMarkdownLanguage;
        return this;
    }

    public String getContent() {
        return content;
    }

    public String getSender() {
        return sender;
    }

    public String getRecipient() {
        return recipient;
    }

    public String getTimestamp() {
        return timestamp;
    }

    public boolean isHasMarkdownLanguage() {
        return hasMarkdownLanguage;
    }

    public SlackNotification build() {
        return new SlackNotification(this);
    }
}

File: C#/Builder/SlackNotificationBuilder.cs
//namespace Builder;

//public class SlackNotification
//{
//    private string content;
//    private string sender;
//    private string recipient;
//    private string timestamp;
//    private bool hasMarkdownLanguage;

//    internal SlackNotification(SlackNotificationBuilder slackNotificationBuilder)
//    {
//        content = slackNotificationBuilder.GetContent();
//        sender = slackNotificationBuilder.GetSender();
//        recipient = slackNotificationBuilder.GetRecipient();
//        timestamp = slackNotificationBuilder.GetTimestamp();
//        hasMarkdownLanguage = slackNotificationBuilder.IsHasMarkdownLanguage();
//    }

//    public string Content => content;
//    public string Sender => sender;
//    public string Recipient => recipient;
//    public string Timestamp => timestamp;
//    public bool HasMarkdownLanguage => hasMarkdownLanguage;
//}

//public class SlackNotificationBuilder : INotificationBuilder
//{
//    private string content;
//    private string sender;
//    private string recipient;
//    private string timestamp;
//    private bool hasMarkdownLanguage;

//    public SlackNotificationBuilder SetContent(string content)
//    {
//        content = content;
//        return this;
//    }

//    public SlackNotificationBuilder SetSender(string sender)
//    {
//        sender = sender;
//        return this;
//    }

//    public SlackNotificationBuilder SetRecipient(string recipient)
//    {
//        recipient = recipient;
//        return this;
//    }

//    public SlackNotificationBuilder SetTimestamp(string timestamp)
//    {
//        timestamp = timestamp;
//        return this;
//    }

//    public SlackNotificationBuilder SetMarkdownLanguage(bool hasMarkdownLanguage)
//    {
//        hasMarkdownLanguage = hasMarkdownLanguage;
//        return this;
//    }

//    internal string GetContent() => content;
//    internal string GetSender() => sender;
//    internal string GetRecipient() => recipient;
//    internal string GetTimestamp() => timestamp;
//    internal bool IsHasMarkdownLanguage() => hasMarkdownLanguage;

//    public SlackNotification Build()
//    {
//        return new SlackNotification(this);
//    }
//}
